test/test_linalg.py

```diff
@@ -1903,6 +1903,7 @@ def run_test_case(input, ord, dim, keepdim):
                 input_numpy = input.cpu().numpy()
                 result_numpy = np.linalg.norm(input_numpy, ord, dim, keepdim)
                 result = torch.linalg.norm(input, ord, dim, keepdim)
+                self.assertEqual(result.shape, result_numpy.shape, msg=msg)
```

Comment:
Interestingly I faced exactly the same problem with `torch.linalg.cond` in https://github.com/pytorch/pytorch/pull/58041.
Maybe `self.assertEqual(torch.tensor, numpy.ndarray)` should check the shape as well.

Reply:
I believe #59067 will add a shape check, and that's how @pmeier found this issue

Reply:
> I believe #59067 will add a shape check

Kinda. It was already there, but never invoked. Now that #59067 has landed, you can remove this check and update the PR.